api/v1beta1/cluster_types.go

```diff
@@ -758,6 +761,7 @@ type MachinePoolTopology struct {
 	// failureDomains is the list of failure domains the machine pool will be created in.
 	// Must match a key in the FailureDomains map stored on the cluster object.
 	// +optional
+	// +kubebuilder:validation:MaxItems=1000
```

Comment:
```suggestion
	// +kubebuilder:validation:MaxItems=100
```
Should be good as well IMO